### PR TITLE
Use the newly introduced feature for conf node inclusion and remove some trivial conf files

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@jsarha @RanderWang pls review.

**Review by RanderWang — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

File: tools/topology/topology2/platform/intel/hdmi-generic.conf [hunk 1]
```diff
@@ -125,6 +166,23 @@ Object.PCM {
 		}
 		direction playback
 	}
+	# Include HDMI4 PCM conditionally
+	IncludeByKey.NUM_HDMIS {
+	"4"	{
+			pcm.8 {
```
Comment (lyakh) [inline tools/topology/topology2/platform/intel/hdmi-generic.conf:172]: nitpick: an alternative indentation could be:
```

	IncludeByKey.NUM_HDMIS {
		"4" {
			pcm.8 {
```

File: tools/topology/topology2/platform/intel/alh-2nd-spk.conf [hunk 2]
```diff
@@ -1,32 +0,0 @@
-
```
Comment (kv2019i) [inline tools/topology/topology2/platform/intel/alh-2nd-spk.conf:1]: Would be helpful for mention the dependency to alsa-utils. E.g. https://github.com/alsa-project/alsa-utils/pull/187 and/or alsa-utils version (which is harder as no released version has this).

**Review by kv2019i — APPROVED** [review]
Nice improvement @ranj063 and @aiChaoSONG . This looks easier to read.

Minor comment on mentioning alsa-utils dependencies. I think we should systematically mention dependencies where possible. alsa-utils is not the easiest package to upgrade as it's an essential system package, so risk of misconfiguration and/or missed dependencies is very, very high.

**Pull request merged**